pkgs/top-level/all-packages.nix

```diff
@@ -41186,4 +41184,22 @@ with pkgs;
   insulator2 = callPackage ../applications/misc/insulator2 {};
 
   animdl = python3Packages.callPackage ../applications/video/animdl { };
+
+  ### BY-NAME
+
+  duckstation = callPackage ../by-name/du/duckstation/package.nix {
```

Comment:
Is there any particular reason why it's been moved to the bottom of the file?
(Asking because `shaderc-changes.patch` is also a thing on PCSX2 (#305509), and I didn't know about `excludes`, so I'm wondering whether to do the exact same thing with PCSX2.)

Reply:
This shouldn't be moved to the bottom and the overwrite should also really be in the package file

Reply:
> Is there any particular reason why it's been moved to the bottom of the file?

By personal choice.
Since the old hierarchy is deprecated, but it can still be needed in things like these, it is cleaner to segregate the old hierarchy from the new.

Also, we know well that `all-packages.nix` is too prone to merge conflicts sometimes, and this reduces conflicts a little bit.

> (Asking because `shaderc-changes.patch` is also a thing on PCSX2 (#305509), and I didn't know about `excludes`, so I'm wondering whether to do the exact same thing with PCSX2.)

This is way better than vendoring a modified patch ourselves, given that this patch can be fetched instead.

> and the overwrite should also really be in the package file

I  believe it betrays the intention of the input parameters in the package file. If I am passing the unpatched `shaderc` as an input library, it makes little sense to modify it internally and recompile it.
It is more natural to pass the patched `shaderc` as input instead.

(We can call this "one package, one build step".)

Reply:
There's a pro to modify `shaderc` in the package.

Let's say someone wants to override Duckstation's input packages (including `shaderc`), maybe because they want to compile them with some optimizations or something.

If the `shaderc` override is done in `all-packages`, then the user must also manually re-add the patches.

If instead it's done in the package, there's no such problem.
(The problem is in case they wanted to remove the patches.)

Reply:
> If the shaderc override is done in all-packages, then the user must also manually re-add the patches.

You are arguing that we should override _inside_ because of a hypothetical someone that wants to override _outside_. It looks self-defeating.

Also, if the person is interested in override shaderc as you suggest, what are the chances that this overridden shaderc would not be compatible with the patches we did inside the expression?

Reply:
> Also, we know well that all-packages.nix is too prone to merge conflicts sometimes, and this reduces conflicts a little bit.

Actually appending to the end of the file is super prone to merge conflicts and it was probably done one purpose to not create a new section. Please just leave the callPackage statement where it is.

Reply:
OK, let's rework this.